h2o-algos/src/main/java/hex/tree/xgboost/XGBoostUpdateTask.java

```diff
@@ -0,0 +1,121 @@
+package hex.tree.xgboost;
+
+import ml.dmlc.xgboost4j.java.*;
+import water.H2O;
+import water.MRTask;
+import water.util.IcedHashMapGeneric;
+
+import java.io.*;
+import java.util.*;
+
+public class XGBoostUpdateTask extends MRTask<XGBoostUpdateTask> {
+
+    private final XGBoostModelInfo _sharedmodel;
+    private final XGBoostOutput _output;
+    transient Booster booster;
+    private byte[] rawBooster;
+
+    private final XGBoostModel.XGBoostParameters _parms;
+    private final int tid;
+
+    private IcedHashMapGeneric.IcedHashMapStringString rabitEnv = new IcedHashMapGeneric.IcedHashMapStringString();
+
+    XGBoostUpdateTask(Booster booster,
+                      XGBoostModelInfo inputModel,
+                      XGBoostOutput _output,
+                      XGBoostModel.XGBoostParameters _parms,
+                      int tid,
+                      Map<String, String> workerEnvs) {
+        this._sharedmodel = inputModel;
+        this._output = _output;
+        this._parms = _parms;
+        this.tid = tid;
+        this.rawBooster = XGBoost.getRawArray(booster);
+        rabitEnv.putAll(workerEnvs);
+    }
+
+    @Override
+    protected void setupLocal() {
+        try {
+            update();
+        } catch (XGBoostError xgBoostError) {
+            xgBoostError.printStackTrace();
+        }
+    }
+
+    private void update() throws XGBoostError {
+        HashMap<String, Object> params = XGBoostModel.createParams(_parms, _output);
+
+        String[] featureMap = new String[]{""};
+
+        rabitEnv.put("DMLC_TASK_ID", String.valueOf(H2O.SELF.index()));
+
+        // DON'T put this before createParams, createPrams calls train() which isn't supposed to be distributed
+        // just to check if we have GPU on the machine
+        Rabit.init(rabitEnv);
+
+        DMatrix trainMat = XGBoostUtils.convertFrameToDMatrix(
+                _sharedmodel._dataInfoKey,
+                _fr,
+                true,
+                _parms._response_column,
+                _parms._weights_column,
+                _parms._fold_column,
+                featureMap,
+                _output._sparse);
+
+        if(null == trainMat) {
+            return;
+        }
+
+        // For feature importances - write out column info
+        OutputStream os;
+        try {
+          os = new FileOutputStream("/tmp/featureMap.txt");
```

Comment:
This needs to be parameterized by at least by job name else we will end with competing overwrites of the file.